MonitorAlert/run.ps1

```diff
@@ -0,0 +1,150 @@
+    * Update to add support for the common alert schema
+#>
+
+using namespace System.Net
+
+# Input bindings are passed in via param block.
+param($Request, $TriggerMetadata)
+
+# Write to the Azure Functions log stream.
+Write-Host "PowerShell HTTP trigger function processed a request."
+
+function EncodeSlackHtmlEntities {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $ToEncode
+    )
+
+    # Encode entities as defined in https://api.slack.com/docs/message-formatting
+
+    $encoded = $ToEncode.Replace("&", "&amp;"). `
+        Replace("<", "&lt;"). `
+        Replace(">", "&gt;")
+
+    return $encoded
+}
+
+function New-SlackMessageFromAlert
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $channel,
+        [Parameter(Mandatory=$true)]
+        [object] $alert
+    )
+
+    $alertAttachmentColours = @{
+        "Activated" = "#ff0000"
+        "Resolved" = "#00a86b"
+    }
+    
+    $lowerAlertStatus = $alert.status.ToLower()
+    
+    $encodedResourceName = EncodeSlackHtmlEntities -ToEncode $alert.context.resourceName
+    $encodedRuleName = EncodeSlackHtmlEntities -ToEncode $alert.context.name
+    $encodedMetricName = EncodeSlackHtmlEntities -ToEncode $alert.context.condition.metricName
+    $encodedMetricOperator = EncodeSlackHtmlEntities -ToEncode $alert.context.condition.operator
+    $encodedMetricThreshhold = EncodeSlackHtmlEntities -ToEncode $alert.context.condition.threshold
+    $portalLink = $alert.context.portalLink
+    
+    $slackMessage = @{ 
+        channel = "#$($channel)"
+        attachments = @(
+            @{
+                color = $alertAttachmentColours[$alert.status]
+                "title_link" = $portalLink
+                title = "Alert $($lowerAlertStatus) for $($encodedResourceName)"
+                text = "The following metric rule belonging to $($encodedRuleName) has been $($lowerAlertStatus):`n`n$($encodedMetricName) $($encodedMetricOperator) $($encodedMetricThreshhold)`n`nPlease visit the resource <$($portalLink)|in the portal> for more information."
+            }
+        ) 
+    }
+
+    return $slackMessage
+}
+
+function Push-OutputBindingWrapper 
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [HttpStatusCode] $Status,
+        [Parameter(Mandatory=$true)]
+        [string] $Body
+    )
+
+    Push-OutputBinding -Name Response -Value ([HttpResponseContext]@{ 
+        StatusCode = $Status
+        Body = $Body
+    })
+}
+
+function Send-MessageToSlack 
+{
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $slackToken,
+        [Parameter(Mandatory=$true)]
+        [hashtable] $message
+    )
+
+    $serializedMessage = "payload=$($message | ConvertTo-Json)"
+
+    Invoke-RestMethod -Uri https://hooks.slack.com/services/$($slackToken) -Method POST -UseBasicParsing -Body $serializedMessage
+}
+
+# Parse query and get settings from environment variables
+$channel = $Request.Query.Channel
```

Comment:
Does this need a try catch to throw a useful error if Channel isn't present?

Reply:
If the channel isn't present,  it returns a null and is caught by some validation further down and returns "BadRequest" http status code to the client with info on how to resolve it